src/GraphQL/Execution/DocumentExecuter.cs

```diff
@@ -298,18 +292,14 @@ await listener.AfterExecutionAsync(context)
                     ex = exceptionContext.Exception;
                 }
 
-                result = new ExecutionResult
-                {
-                    Errors = new ExecutionErrors
-                    {
-                        ex is ExecutionError executionError ? executionError : new UnhandledError(exceptionContext?.ErrorMessage ?? "Error executing document.", ex)
-                    }
-                };
+                (result ??= new ExecutionResult()).AddError(ex is ExecutionError executionError ? executionError : new UnhandledError(exceptionContext?.ErrorMessage ?? "Error executing document.", ex));
```

Comment:
And, this is now harder to read than it was before.

Reply:
My thought was that errors from the context should complement the errors already recorded in the result, and not replace them. This is not true? What if `result` is not null?

Reply:
Ok.  Currently that is only possible if a document listener throws an error after an execution.

Reply:
Indeed.